### DISCUSSION TITLE
deploying-asp-net-core-applications-to-kubernetes-part-12-tips-tricks-and-edge-cases/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#344] [Announcements]
# deploying-asp-net-core-applications-to-kubernetes-part-12-tips-tricks-and-edge-cases/

In this post I describe some of the smaller pieces of advice, tips, and info I found when running applications on Kubernetes

https://andrewlock.net/deploying-asp-net-core-applications-to-kubernetes-part-12-tips-tricks-and-edge-cases/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 20 2020, 03:58, in reply to [fredhusser](https://disqus.com/by/disqus_oLJiFqhfRv/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/344#discussioncomment-6059638)</em>

---

<p>Thanks, glad you found it useful :)</p><p>I'm not 100% sure on how it works with Azure Key Vault specifically. There are mechanisms (generally) for avoiding that by using managed identities/roles. i.e. the vault can be accessed by virtue of the request coming from a "trusted" host, i.e. the kubernetes cluster.</p><p>It's a well known chicken and egg problem though, and the proposed solutions to it are often cumbersome and require hooking in to the infrastructure.</p><p>The typical approach we use steps outside the "native k8s" approach, and relies on machine roles with specific access to AWS Secrets Manager, to pull down those secrets on app startup, as I described in a previous post <a href="https://andrewlock.net/secure-secrets-storage-for-asp-net-core-with-aws-secrets-manager-part-2/" rel="nofollow noopener" title="https://andrewlock.net/secure-secrets-storage-for-asp-net-core-with-aws-secrets-manager-part-2/">https://andrewlock.net/secure-secrets-storage-for-asp-net-core-with-aws-secrets-manager-part-2/</a>. It's not the most elegant, but it mostly works well enough for us!</p>

[a comment by an automated account is not shown]

**Comment by akhanalcs** [comment]
Excellent series. I followed along all your posts in this series and learned a lot.  
Can you please do .NET Aspire + Radius series now?

**Comment by akhanalcs** [comment]
>There are a few subtle gotchas with configuring the data-protection system in a clustered environment. I'll write a separate post(s) detailing our approach, as there are a few things to consider.

Do you have a blog post on this that we can read up on? Thank you!
